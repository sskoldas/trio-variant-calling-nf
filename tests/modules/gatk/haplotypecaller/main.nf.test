nextflow_process {

    name "Test Process GATK_HAPLOTYPECALLER"
    script "modules/gatk/haplotypecaller/main.nf"
    process "GATK_HAPLOTYPECALLER"

    test("Should call son's haplotype correctly") {

        setup {
            run("SAMTOOLS_INDEX") {
                script "modules/samtools/index/main.nf"
                process {
                    """
                    input[0] = file("${projectDir}/data/bam/reads_son.bam")
                    """ 
                }
            }
        }

        when {
            params {
                outdir = "tests/results"
            }
            process {
                """
                input[0] = SAMTOOLS_INDEX.out
                input[1] = file("${projectDir}/data/ref/ref.fasta")
                input[2] = file("${projectDir}/data/ref/ref.fasta.fai")
                input[3] = file("${projectDir}/data/ref/ref.dict")
                input[4] = file("${projectDir}/data/ref/intervals.bed")
                """
            }
        }

        then {
            assert process.success
            assert path(process.out[0][0]).readLines().contains('#CHROM	POS	ID	REF	ALT	QUAL	FILTER	INFO	FORMAT	reads_son')
        }

    }

}
