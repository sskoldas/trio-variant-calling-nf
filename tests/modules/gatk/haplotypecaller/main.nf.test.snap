{
    "Should call son's haplotype correctly": {
        "content": [
            {
                "0": [
                    "reads_son.bam.g.vcf:md5,429406f5339e4d36691164be3ddfc27f"
                ],
                "1": [
                    "reads_son.bam.g.vcf.idx:md5,7486a1741f1e657c1bd3d601832751a2"
                ],
                "idx": [
                    "reads_son.bam.g.vcf.idx:md5,7486a1741f1e657c1bd3d601832751a2"
                ],
                "vcf": [
                    "reads_son.bam.g.vcf:md5,429406f5339e4d36691164be3ddfc27f"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-09-28T00:06:15.627800973"
    }
}